### ISSUE TITLE
ballerina compile fails for module having custom package in ballerina tests

### ISSUE THREAD

**Issue by azinneera** [ballerina-platform/ballerina-lang#19558]
**Description:**
Running the ballerina compile command for a module fails when there are tests written using a custom package.

The following error gets printed in the ballerina-internal.log file.
`[2019-10-24 11:59:31,130] SEVERE {b7a.log.crash} - null 
java.lang.NullPointerException
	at org.wso2.ballerinalang.programfile.PackageFileWriter.writePackage(PackageFileWriter.java:51)
	at org.ballerinalang.packerina.writer.BirFileWriter.writeBIRToPath(BirFileWriter.java:100)
	at org.ballerinalang.packerina.task.CreateBirTask.writeImportBir(CreateBirTask.java:73)
	at org.ballerinalang.packerina.task.CreateBirTask.execute(CreateBirTask.java:53)
	at org.ballerinalang.packerina.TaskExecutor.executeTasks(TaskExecutor.java:38)
	at org.ballerinalang.packerina.cmd.BuildCommand.execute(BuildCommand.java:395)
	at java.util.Optional.ifPresent(Optional.java:159)
	at org.ballerinalang.tool.Main.main(Main.java:59)`

**Link to the sample test bal file and balo**
https://gist.github.com/azinneera/6b1f5c5e450c2310a59e26d16697141e

**Steps to reproduce:**
Write a custom package and refer it in the test bal file.
Run ballerina build command to the module

**Suggested Labels (optional):**
Bug

**Comment by azinneera** [comment]
When the .bir file is copied to the ~/.ballerina/bir_cache and then when the tests are run, the following error is getting logged on the ballerina-internal.log 

```
[2019-10-28 15:58:45,052] SEVERE {b7a.log.crash} - org.ballerinalang.util.exceptions.BallerinaException: java.lang.NullPointerException 
org.ballerinalang.util.exceptions.BallerinaException: org.ballerinalang.util.exceptions.BallerinaException: java.lang.NullPointerException
	at org.ballerinalang.testerina.core.BTestRunner.formatErrorMessage(BTestRunner.java:819)
	at org.ballerinalang.testerina.core.BTestRunner.lambda$null$28(BTestRunner.java:676)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at org.ballerinalang.testerina.core.BTestRunner.lambda$execute$34(BTestRunner.java:669)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at org.ballerinalang.testerina.core.BTestRunner.execute(BTestRunner.java:638)
	at org.ballerinalang.testerina.core.BTestRunner.runTest(BTestRunner.java:160)
	at org.ballerinalang.testerina.util.TesterinaUtils.executeTests(TesterinaUtils.java:132)
	at org.ballerinalang.packerina.task.RunTestsTask.execute(RunTestsTask.java:81)
	at org.ballerinalang.packerina.TaskExecutor.executeTasks(TaskExecutor.java:38)
	at org.ballerinalang.packerina.cmd.TestCommand.execute(TestCommand.java:276)
	at java.util.Optional.ifPresent(Optional.java:159)
	at org.ballerinalang.tool.Main.main(Main.java:59)
Caused by: org.ballerinalang.util.exceptions.BallerinaException: java.lang.NullPointerException
	at org.ballerinalang.testerina.core.entity.TesterinaFunction.runOnSchedule(TesterinaFunction.java:146)
	at org.ballerinalang.testerina.core.entity.TesterinaFunction.invoke(TesterinaFunction.java:66)
	at org.ballerinalang.testerina.core.BTestRunner.lambda$null$28(BTestRunner.java:672)
	... 11 more
Caused by: java.lang.NullPointerException
	at sun.net.util.URLUtil.urlNoFragString(URLUtil.java:50)
	at sun.misc.URLClassPath.getLoader(URLClassPath.java:512)
	at sun.misc.URLClassPath.getNextLoader(URLClassPath.java:484)
	at sun.misc.URLClassPath.getResource(URLClassPath.java:238)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:365)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at asma.pet-be.tests.pet-be-test.setup(tests/pet-be-test.bal:27)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.ballerinalang.testerina.core.entity.TesterinaFunction.lambda$runOnSchedule$0(TesterinaFunction.java:130)
	at org.ballerinalang.jvm.scheduling.SchedulerItem.execute(Scheduler.java:496)
	at org.ballerinalang.jvm.scheduling.Scheduler.run(Scheduler.java:238)
	at org.ballerinalang.jvm.scheduling.Scheduler.runSafely(Scheduler.java:208)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by Kishanthan** [comment]
The original issue in the description is fixed already with 1.0.2. Please check https://github.com/ballerina-platform/ballerina-lang/issues/19303.

**Comment by azinneera** [comment]
@Kishanthan, yes fix for #19303 resolves the issue. Thank you for pointing it out. We may close this issue.
